tests/perf/pubsub_bulk_publish_http/pubsub_bulk_publish_http_test.go

```diff
@@ -0,0 +1,148 @@
+//go:build perf
+// +build perf
+
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub_bulk_publish_http
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/dapr/dapr/tests/perf/utils"
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/dapr/dapr/tests/runner/loadtest"
+	"github.com/dapr/dapr/tests/runner/summary"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	tr *runner.TestRunner
+)
+
+const (
+	k6AppName = "k6-test-app"
+	topicName = "bulkpublishperftopic"
+)
+
+type testCase struct {
+	broker        string
+	publishType   string
+	topic         string
+	bulkSize      int
+	messageSizeKb int
+	durationMs    int
+	numVus        int
+}
+
+var brokers = []kube.ComponentDescription{
+	{
+		Name:      "memory-broker",
+		Namespace: &kube.DaprTestNamespace,
+		TypeName:  "pubsub.in-memory",
+		MetaData:  map[string]kube.MetadataValue{},
+		Scopes:    []string{k6AppName},
+	},
+}
+
+var brokersNames string
+
+func init() {
+	brokersList := []string{}
+	for _, broker := range brokers {
+		brokersList = append(brokersList, broker.Name)
+	}
+	brokersNames = strings.Join(brokersList, ",")
+}
+
+func TestMain(m *testing.M) {
+	utils.SetupLogs("pubsub_bulk_publish_http_test")
+
+	tr = runner.NewTestRunner("pubsub_bulk_publish_http", []kube.AppDescription{}, brokers, nil)
+	os.Exit(tr.Start(m))
+}
+
+func TestPubsubBulkPublishHttpPerformance(t *testing.T) {
+	publishTypes := []string{"normal", "bulk"}
+	bulkSizes := []int{10, 100}
```

Comment:
Should we have a separate config file, catering to these kind of configurations? 
Reason being, it will make our perf tests able to run on different kind of scenarios - on need basis - let's say some user wants to test it in a particular scenario.
But, I see that most of these constants till now have been in perf code only. So, if you feel this would be a separate task, may be we can have a separate issue for it.

Reply:
Yes, I think we can do this separately for all perf tests and have a common way to configure and run them on individual forks.